tower-http/src/add_extension.rs

```diff
@@ -1,14 +1,22 @@
+//! Middleware for adding some shareable value to [request extensions].
```

Comment:
nit, take it or leave it: what do you think about making a little more explicit that this uses `Clone`? 
```suggestion
//! Middleware that clones a value into each request's [extensions].
```
or something?